cfgmgr/teammgr.cpp

```diff
@@ -112,18 +112,53 @@ void TeamMgr::doTask(Consumer &consumer)
     }
 }
 
-
 void TeamMgr::cleanTeamProcesses()
 {
     SWSS_LOG_ENTER();
     SWSS_LOG_NOTICE("Cleaning up LAGs during shutdown...");
-    for (const auto& it: m_lagList)
+
+    std::unordered_map<std::string, pid_t> aliasPidMap;
+
+    for (const auto& alias: m_lagList)
+    {
+        std::string res;
+        pid_t pid;
+
+        {
+            std::stringstream cmd;
+            cmd << "cat " << shellquote("/var/run/teamd/" + alias + ".pid");
+            EXEC_WITH_ERROR_THROW(cmd.str(), res);
+
+            pid = static_cast<pid_t>(std::stoul(res, nullptr, 10));
+            aliasPidMap[alias] = pid;
+
+            SWSS_LOG_INFO("Read port channel %s pid %d", alias.c_str(), pid);
+        }
+
+        {
+            std::stringstream cmd;
+            cmd << "kill -TERM " << pid;
+            EXEC_WITH_ERROR_THROW(cmd.str(), res);
+
+            SWSS_LOG_INFO("Sent SIGTERM to port channel %s pid %d", alias.c_str(), pid);
+        }
+    }
+
+    for (const auto& cit: aliasPidMap)
     {
-        //This will call team -k kill -t <teamdevicename> which internally send SIGTERM 
-        removeLag(it);
+        const auto &alias = cit.first;
+        const auto &pid = cit.second;
+
+        std::stringstream cmd;
+        std::string res;
+
+        SWSS_LOG_NOTICE("Waiting for port channel %s to stop...", alias.c_str());
+
+        cmd << "tail -f --pid=" << pid << " /dev/null";
```

Comment:
The idea behind us moving earlier to the approach of using "teamd -k" itself was to do a proper clean and wait for the process is killed ( teamd implementation indirectly calls this API daemon_pid_file_kill_wait(pid, 30sec) )

I see this wait is implemented here with exec/tail, I was thinking of not calling EXEC, and instead use may be use something similar as done here ? Let me know you opinion 

https://github.com/thingnario/libdaemon-0.14/blob/9ad15b1b69bc59ae0752e939c87f600cfb0f4f5c/libdaemon/dpid.c#L177

The way we wait to see if the process exits is by calling kill(pid, 0) which as per the kill man page is 
```
       If sig is 0, then no signal is sent, but existence and permission
       checks are still performed; this can be used to check for the
       existence of a process ID or process group ID that the caller is
       permitted to signal.
```

Reply:
@judyjoseph i have replaced system calls with Signal API calls. Please have a look.